cmd/panfs.go

```diff
@@ -712,23 +731,14 @@ func (fs *PANFSObjects) DeleteBucket(ctx context.Context, bucket string, opts De
 		return toObjectErr(err, bucket)
 	}
 
-	// TODO: delete bucket operation should preserve user objects on the realm
-	if !opts.Force {
-		// Attempt to delete regular bucket.
-		if err = removePanFSBucketDir(ctx, bucketDir); err != nil {
-			return toObjectErr(err, bucket)
-		}
-	} else {
-		// Still using here .minio.sys as temp dir for delete bucket
-		tmpBucketPath := pathJoin(fs.fsPath, minioMetaTmpBucket, bucket+"."+mustGetUUID())
-		if err = Rename(bucketDir, tmpBucketPath); err != nil {
-			return toObjectErr(err, bucket)
-		}
-
-		go func() {
-			fsRemoveAll(ctx, tmpBucketPath) // ignore returned error if any.
-		}()
+	// only remove .s3 directory
+	deletePath := path.Join(fs.fsPath, minioMetaTmpBucket, bucket+"."+mustGetUUID())
+	if err = Rename(path.Join(bucketDir, panfsMetaDir), deletePath); err != nil {
+		return toObjectErr(err, bucket)
 	}
+	go func() {
+		fsRemoveAll(ctx, deletePath)
+	}()
```

Comment:
a function in a separate goroutine may not have time to execute before the end of the main function.

Reply:
Any other ways except synchronous delete then?

Reply:
you can use waitGroup, for example:
wg.Add(1)
go func(){
...
wg.Done
}()
wg.Wait()

this is the simplest example.